site/ar/tutorials/text/nmt_with_attention.ipynb

```diff
@@ -0,0 +1,1136 @@
+    "\n",
+    "import unicodedata\n",
+    "import re\n",
+    "import numpy as np\n",
+    "import os\n",
+    "import io\n",
+    "import time"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "wfodePkj3jEa"
+   },
+   "source": [
+    "## <div dir=\"rtl\"> تنزيل وتحضير مجموعة البيانات </div>\n",
+    "\n",
+    "<div dir=\"rtl\">\n",
+    "\n",
+    "سنستخدم مجموعة بيانات لغوية مقدمة من موقع\n",
+    "[manythings.org](http://www.manythings.org/anki/).\n",
+    " تحتوي هذه المجموعة على أزواج لغوية في الشكل التالي:\n",
+    "\n",
+    "</div>\n",
+    "\n",
+    "```\n",
+    "May I borrow this book?\t¿Puedo tomar prestado este libro?\n",
+    "```\n",
+    "<div dir=\"rtl\">\n",
+    "\n",
+    "هناك مجموعة متنوعة من اللغات المتاحة ، ولكننا سنستخدم مجموعة البيانات الإنجليزية الإسبانية. لتبسيط الأمر ، اضفنا نسخة من مجموعة البيانات هذه على Google Cloud ، ولكن يمكنك أيضًا تنزيل نسختك الخاصة. بعد تنزيل مجموعة البيانات ، إليك الخطوات التي سنتخذها لإعدادها:\n",
+    "\n",
+    "1. أضف رمز *start* و *end* لكل جملة.\n",
+    "2. نظف الجمل بإزالة [الأحرف الخاصة](https://owasp.org/www-community/password-special-characters).\n",
+    "3.كوِّن فهرس الكلمات وفهرس الكلمات العكسي (قواميس تعطي لكل كلمة معرف وحيد والعكس بالعكس).\n",
+    "4. كمل كل جملة إلى الطول الأقصى.\n",
+    "\n",
+    "\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "kRVATYOgJs1b"
+   },
+   "outputs": [],
+   "source": [
+    "# Download the file\n",
+    "path_to_zip = tf.keras.utils.get_file(\n",
+    "    'spa-eng.zip', origin='http://storage.googleapis.com/download.tensorflow.org/data/spa-eng.zip',\n",
+    "    extract=True)\n",
+    "\n",
+    "path_to_file = os.path.dirname(path_to_zip)+\"/spa-eng/spa.txt\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "rd0jw-eC3jEh"
+   },
+   "outputs": [],
+   "source": [
+    "# Converts the unicode file to ascii\n",
+    "def unicode_to_ascii(s):\n",
+    "  return ''.join(c for c in unicodedata.normalize('NFD', s)\n",
+    "      if unicodedata.category(c) != 'Mn')\n",
+    "\n",
+    "\n",
+    "def preprocess_sentence(w):\n",
+    "  w = unicode_to_ascii(w.lower().strip())\n",
+    "\n",
+    "  # creating a space between a word and the punctuation following it\n",
+    "  # eg: \"he is a boy.\" => \"he is a boy .\"\n",
+    "  # Reference:- https://stackoverflow.com/questions/3645931/python-padding-punctuation-with-white-spaces-keeping-punctuation\n",
+    "  w = re.sub(r\"([?.!,¿])\", r\" \\1 \", w)\n",
+    "  w = re.sub(r'[\" \"]+', \" \", w)\n",
+    "\n",
+    "  # replacing everything with space except (a-z, A-Z, \".\", \"?\", \"!\", \",\")\n",
+    "  w = re.sub(r\"[^a-zA-Z?.!,¿]+\", \" \", w)\n",
+    "\n",
+    "  w = w.rstrip().strip()\n",
```

Comment:
As I suggested in this PR, https://github.com/tensorflow/docs/pull/1512

This
```
w = w.rstrip().strip()
```

is equivalent to this

```
w = w.strip()
```